server/game/Card.js

```diff
@@ -516,12 +527,16 @@ class Card extends EffectSource {
         });
     }
 
+    canPlayAsUpgrade() {
+        return this.anyEffect('canPlayAsUpgrade') || this.type === 'upgrade';
+    }
+
     /**
      * Checks whether the passed card meets the upgrade restrictions (e.g.
      * Opponent cards only, specific factions, etc) for this card.
      */
     canAttach(card, context) { // eslint-disable-line no-unused-vars
-        return card && card.getType() === 'creature' && this.getType() === 'upgrade';
+        return card && card.getType() === 'creature';
```

Comment:
I don't like that you're removing the safety check here.  Non-upgrades shouldn't be able to be attached to creatures under normal circumstances....

Reply:
Maybe you can check for `this.anyEffect('canPlayAsUpgrade')` and `context.ability.isCardPlayed()`?

Reply:
Oops, I think I mistakenly removed it while trying to clean up the rest of the file.